src/vuex/modules/index/store.ts

```diff
@@ -145,40 +158,113 @@ const actions = createActions({
       throw new Error(`Collection "${name}" already exist`)
     }
 
-    commit.setLoadingCollectionsForIndex({ index, loading: true })
+    commit.setLoadingCollections({ index, loading: true })
 
     const collectionType = isRealtime
-      ? CollectionType.STORED
-      : CollectionType.REALTIME
+      ? CollectionType.REALTIME
+      : CollectionType.STORED
+
     let collection = new Collection(name, collectionType)
+
+    if (isRealtime) {
+      commit.addCollection({ index, collection })
+      commit.setLoadingCollections({ index, loading: false })
+      return
+    }
+
     collection.mapping = mapping
+    collection.dynamic = dynamic
 
     await rootGetters.kuzzle.$kuzzle.collection.create(index.name, name, {
+      dynamic,
       properties: {
         ...mapping
-      },
-      dynamic
+      }
     })
 
     commit.addCollection({ index, collection })
-    commit.setLoadingCollectionsForIndex({ index, loading: false })
+    commit.setLoadingCollections({ index, loading: false })
+  },
+  async updateCollection(
+    context,
+    { index, name, isRealtime, mapping, dynamic }: UpdateCollectionPayload
+  ) {
+    const { commit, rootGetters } = indexActionContext(context)
+
+    if (!index.doesCollectionExist(name)) {
+      throw new Error(`Collection "${name}" doesn't exist`)
+    }
+
+    if (isRealtime) {
+      return Promise.resolve()
+    }
+
+    const collectionType = isRealtime
+      ? CollectionType.STORED
+      : CollectionType.REALTIME
+
+    let updatedCollection = new Collection(name, collectionType)
+
+    updatedCollection.mapping = mapping
+    updatedCollection.dynamic = dynamic
+
+    await rootGetters.kuzzle.$kuzzle.query({
```

Comment:
Why do we use `query` here? Is this compatible btw v1 and v2?

Reply:
I had first tested with the dedicated method of the SDK but for some reason it didn't work (maybe related to `kuzzleWrapper`). Since it works very well with `query`, my plan is to add another TODO and deal with the problem in another PR